### DISCUSSION TITLE
Haos script does not work on Proxmox 7.3-6

### DISCUSSION THREAD

**Discussion by phulstaert** [tteck/Proxmox#1212] [Help Needed]
I cannot get Haos to run on my new Proxmox 7.3-6 machine.

Tried both a new install of Haos as a restore from a backup of a working Haos install on a Proxmox 7.3-4 installation.
Both times I get the same symptom.

![image](https://user-images.githubusercontent.com/8971696/226926135-646dd7b2-ec2a-4682-a7ac-f30191bb3615.png)

Settings:

![image](https://user-images.githubusercontent.com/8971696/226926222-2e8249ce-5f45-41ae-ae6a-39508182c640.png)

![image](https://user-images.githubusercontent.com/8971696/226926312-86cacde6-e9d1-467d-bbd7-bc5945a4d3ec.png)

On my working Proxmox 7.3-4 install these are my settings:

![image](https://user-images.githubusercontent.com/8971696/226926498-6365c2f7-641e-4b82-a14b-13b1abf29d72.png)

![image](https://user-images.githubusercontent.com/8971696/226926559-d90bcb90-c01b-4d50-bd35-f83869365ebb.png)

I can see under OS Type there is Linux 6.x vs Linux 5.x; but I do not have the Linux 5.x option anymore under Proxmox 7.3-6.
I did try Linux 2.6; none; etc... All with no effect.

I have no clue what is going on here...

**Comment by tteck** [comment]
Ensure that your machine's BIOS settings have UEFI Boot and Virtualization enabled, while Secure Boot is disabled, especially if this is the first time you are running Proxmox on the machine.

**Reply by phulstaert** [reply]
The original Proxmox 7.3-4 runs on a "normal" desktop.

The Proxmox 7.3-6 runs on a Synology DS1522+ as a virtual machine. This device supports nested virtualization, so this should work.
UEFI boot and Virtualisation is enabled. I need to check the Secure boot part...

The VM where Haos runs in was made with the script, so UEFI, secure boot and such are all ok.

Do you mean the host or the VM when you said "your machine"?

**Reply by phulstaert** [reply]
I have noticed that when I disable KVM Hardware Virtualization, it boots!
So, probably an issue with the nested virtualization.
BUT... when booting, it stops before the last step.

![image](https://user-images.githubusercontent.com/8971696/227758273-c8f1fa99-df6e-4a65-865d-853143f4147b.png)

I guess this isn't a script issue anymore, but I would appreciate input or a guess if you have one...
There were no errors in the booting process other than it took a bit longer (2 min to boot - still not ridiculous)

**Comment by tteck** [comment]
While it's technically possible to install Proxmox as a virtual machine on a Synology DS1522+, it's far from an ideal solution. In fact, it's highly questionable whether it would be efficient or reliable at all. 
The DS1522+ is primarily designed for file storage and sharing, and running a hypervisor like Proxmox on it would strain its limited resources. Furthermore, attempting to run both Proxmox and the storage system on the same device could lead to conflicts and potentially catastrophic data loss. 
IMHO, anyone considering this setup would be wise to think again and explore more appropriate solutions.

**Reply by OddSquirrel** [reply]
Solid advice. Granted, I run my Proxmox Backup Server in a VM on a Synology NAS, but that thing basically idles along all day between 0 and 0.2% CPU use and only gets to work for a handful of minutes late at night when it's got nothing else to do anyway.

While the software is great, Synology NAS are very anemic when it comes to hardware. I'd trust a QNAP to be able to run a PVE node (they go up to i9 CPUs), but then again, I don't trust their software. I think scrapped PCs are a great alternative. Whenever I get a new PC, I turn the predecessor into a PVE node, problem solved.

**Reply by phulstaert** [reply]
I appreciate your answer, but it doesn't help me at all.

So you are saying that this Synology product, that is put in the market specifically for the home and small office usage to do File storage, security camera recorder, docker, virtualization and a lot of build in services like mail server, web server, reverse proxy, firewall, DHCP, etc...; is not suitable for virtualization.

Just to make sure,we are talking about the same thing here: 
It has an AMD R1600 - according to AMD this is for the embedded server market and supports both AMD-V and SVM. 
It supports up to 32 GB of memory (this is installed in my system) and has hardware support for SSE4a, SSE4.1, SSE4.2, AVX2 and FMA3.
With its 2 cores and 4 threads at 2.6Ghz / 3.1Ghz, it is comparable in Geekbench 5 single-core to an Intel Core i7 3615QM and multicore to an Intel Core 2 Quad Q9400.
At the moment, my system is recording 6 HD camera's and running 2 VMs where one is Proxmox and this one is running 5 VMs successfully with only ~5% CPU usage with an IO delay of 0.0% - 0.23%. I do have to say I have installed 2x 512 GB of NVMe SSD read and write cache on this Synology machine, and it has 24 TB of HDD installed. This all is connected with a 4Gbps link to the network.

My original question was why Haos did not start - but you say the Synology is the issue - although the 5 other VMs don't have any issue with it... Yet I never get to hear just exactly what the issue is and why Synology should be an issue.
To me, Synology products have an excellent beginner GUI, and they give the opportunity to the mid and advanced users to configure anything they want using the shell.

Also, why is it that "attempting to run both Proxmox and the storage system on the same device could lead to conflicts and potentially catastrophic data loss."
Are you saying that Synology, but also the makers of systems like TrueNAS and FreeNAS don't know what they are doing?
Why are you coming to this conclusion? 

To me, the systems don't have to have a 99.9999% uptime and the weekly backups and daily snapshots that are saved on the attached USB3 SSD are more than enough for backup purposes. 

The only question I have was: 
- why - from a technical standpoint - is it that the VM would not start when KVM hardware virtualization is on.
- why - from a technical standpoint - would the HA CLI not start when the KVM hardware virtualization is off.

Performance is not the issue, I would assume, since a first generation Raspberry Pi with Haos on a crappy SD card has far less performance than this system.

**Reply by OddSquirrel** [reply]
I don't know what else to tell you. Just because Synology's marketing department wrote on a flyer that it can basically host the Internet doesn't make it true.

On the contrary, no other NAS manufacturer will go out of their way like Synology and search high and low for a solution that will just barely fit the minimum criteria. The new NAS that are so great for the home market don't even sport a gfx unit, making them basically unusable for stuff like Plex et al.

Of course, they're still usable for small office use. Especially for file sharing. We like to use their calendar, contacts, chat and other integrated solutions as well. Or the aforementioned PBS. All fine and dandy.

But running hypervisor upon hypervisor is an entirely different ballgame. Why wouldn't you just slap it straight onto VM Manager or fire up a Docker container right on the NAS instead of abstracting things further?

**Reply by phulstaert** [reply]
@OddSquirrel 
Let's agree to disagree :-)

I have a DS415play (the series that are for home-use and transcoding video) and it served me very good for many years.
Now I have a DS1522+ (because the DS415play had a 32bit processor without virtualization support - so no docker nor QEMU) and the new one runs everything I need.
I was tempted to buy a j-series, because that is the "cheap"-line, but am glad I did not.

It runs a Plex server in fact :-) Transcoding a 1080p file on the fly btw is no problem, but I usually transcode in advance... Everything feels very snappy and quick.

I had an "old" Intel i7-6700 CPU system next to it for a few docker containers and a few VM's - from when I had the DS415play. But after the energy costs went up so dramatically, I wanted to do all on one machine.  That one "old" machine was responsible for 14% of my total energy consumption at home. I looked at power-saving features, but when running a few VM's and docker instances, those did almost nothing.

For home use and semi-professional use at home, that DS1522+ is a lot of value for the money. Also, those machines never - NEVER - crashed on me. So I do have confidence in the stability of the Synology products.

On that "old" machine, I got used to Proxmox and I like a lot of features of it. Nested virtualization is nothing new and accepting a very small performance penalty, it would keep everything the same for me. 
I don't like the VM interface of Synology and don't want to go to the shell every time I need to do something (I got spoiled by Proxmox :-)).
I just exported the Haos VM to a qcow2 image and imported this into the VMM of Synology. It started up without a problem. I ran both systems next to each other and ran the last ha core update. The "old" machine was first, but the Synology was only 20 seconds behind. 
This is worth saving more than €250 on electricity bills each year for me.

Every situation/reason/motivation is different and every use case is different. As someone that works in IT, I just could not wrap my head around the fact that every other VM works, but Haos did not. This will be one of those things that will keep me up at night trying to figure out what it was...

**Comment by tteck** [comment]
It's also worth noting that running a nested virtualization setup, such as a Proxmox VM inside a Synology VMM VM, may not be fully supported by either platform.

>why - from a technical standpoint - is it that the VM would not start when KVM hardware virtualization is on.

Insufficient resources: KVM hardware virtualization requires additional system resources, such as CPU extensions and memory. If the Synology DS1522+ does not have enough resources available to support both the host system and the VMs, this could cause the VM to fail to start when KVM hardware virtualization is enabled. 

>why - from a technical standpoint - would the HA CLI not start when the KVM hardware virtualization is off.

The Synology DS1522+ uses Virtual Machine Manager (VMM) to create and manage virtual machines. VMM is a built-in hypervisor that runs on top of the DSM operating system, which is a modified version of Linux. Since VMM runs on top of the DSM operating system, it is not strictly a Type 1 hypervisor. Instead, it can be considered a Type 2 hypervisor that uses a Type 1 hypervisor (KVM) to enable virtualization.

Again, **anyone considering this setup would be wise to think again and explore more appropriate solutions.**

**Reply by phulstaert** [reply]
@tteck, thank you for taking the time to write this answer.
I never knew there was a type 1 or 2 hypervisor. I know there were bare metal hypervisors and "hypervisors installed as an application", but now I know the terminology. Thank you for that.

So am I getting it right when I say that Proxmox runs KVM, and this makes it a type 1 hypervisor, even though Proxmox runs on Debian.
Synology uses QEMU and that is a type 2 hypervisor. When Installing Proxmox as a VM on my NAS, KVM cannot do its work because its "bare metal" is emulated. This would explain why KVM hardware virtualization would not work.
It does trigger me why the other VMs work out of the box. Then again, I work in IT long enough to know IT is not always logical :-)

But it does not explain why the HA CLI would not start, while it would boot.

But as you might have read. I found another solution. 
It is still storage and VM on the same machine, but for my use-case this is more than acceptable.

**Comment by tteck** [comment]
It seems like you have already found another solution, so there is no need for further discussion on this matter. Therefore, I will consider this discussion closed.
